db-engine/src/main/java/com/splicemachine/db/iapi/stats/ColumnStatisticsMerge.java

```diff
@@ -88,20 +92,41 @@ public void init() {
      */
     @Override
     public void accumulate(ColumnStatisticsImpl value) throws StandardException {
-        ColumnStatisticsImpl columnStatistics = (ColumnStatisticsImpl) value;
         if (!initialized) {
-            dvd = columnStatistics.getColumnDescriptor();
+            if (value instanceof FakeColumnStatisticsImpl) {
+                columnStatsType = ItemStatistics.Type.FAKE;
+                FakeColumnStatisticsImpl fakeColumnStatistics = (FakeColumnStatisticsImpl) value;
+                nullCount += fakeColumnStatistics.nullCount();
+                totalCount += fakeColumnStatistics.totalCount();
+                cardinality = fakeColumnStatistics.cardinality();
+                initialized = true;
+                return;
+            }
+
+            dvd = value.getColumnDescriptor();
             assert dvd!=null:"dvd should not be null";
             thetaSketchUnion = Sketches.setOperationBuilder().buildUnion();
-            quantilesSketchUnion = ItemsUnion.getInstance(value.getQuantilesSketch().getK(), ((ColumnStatisticsImpl)value).getColumnDescriptor());
+            quantilesSketchUnion = ItemsUnion.getInstance(value.getQuantilesSketch().getK(), value.getColumnDescriptor());
             frequenciesSketch =  dvd.getFrequenciesSketch();
             initialized = true;
         }
-        assert columnStatistics.quantilesSketch !=null && columnStatistics.frequenciesSketch !=null && columnStatistics.thetaSketch !=null && value!=null:"Sketches should not be null";
-        quantilesSketchUnion.update(columnStatistics.quantilesSketch);
-        frequenciesSketch.merge(columnStatistics.frequenciesSketch);
-        thetaSketchUnion.update(columnStatistics.thetaSketch);
-        nullCount =+ columnStatistics.nullCount();
+
+        if (!value.getType().equals(columnStatsType))
+            throw new RuntimeException("Multiple column stats type co-exists!");
+
+        if (value instanceof FakeColumnStatisticsImpl) {
+            FakeColumnStatisticsImpl fakeColumnStatistics = (FakeColumnStatisticsImpl) value;
+            nullCount += fakeColumnStatistics.nullCount();
+            totalCount += fakeColumnStatistics.totalCount();
+            if (cardinality < fakeColumnStatistics.cardinality())
+                cardinality = fakeColumnStatistics.cardinality();
```

Comment:
Why update cardinality only when it is less than the faked value?

Reply:
This function merges stats from two partitions. I simply assume that the distinct values are randomly distributed in the data, so they are overlapping in the two partitions, and the merged number of distinct values will go with the bigger of the two partitions.